File: Reverse Vowels of a String/Reverse_Vowels_of_a_String.py
class Solution:
    def vowel(self, x):
        x = x.lower()
        return x in "aeiou"
    def reverseVowels(self, s: str) -> str:
        s = list(s)
        left, right = 0, len(s) - 1
        while left < right:
            if self.vowel(s[left]) and self.vowel(s[right]):
                s[left], s[right] = s[right], s[left]
                left += 1
                right -= 1
            elif self.vowel(s[left]):
                right -= 1
            else:
                left += 1
        return "".join(s)
        

File: Reverse Vowels of a String/Reverse_Vowels_of_a_String.cpp
class Solution {
public:
    bool vowel(char x) {
        x = tolower(x);
        if(x == 'a' || x == 'e' || x == 'i' || x == 'o' || x == 'u') return true;
        return false;
    }
    string reverseVowels(string s) {
        int left = 0,right = s.size()-1;
        while(left < right) {
            if(vowel(s[left]) && vowel(s[right])) {
                swap(s[left],s[right]);
                left++;
                right--;
            }
            else if(vowel(s[left])) {
                right--;
            }
            else left++;
        }
        return s;
    }
};